static/js/unread_ops.js

```diff
@@ -104,7 +104,7 @@ export function notify_server_message_read(message, options) {
 // If we ever materially change the algorithm for this function, we
 // may need to update notifications.received_messages as well.
 export function process_visible() {
-    if (overlays.is_active() || !notifications.is_window_focused()) {
+    if (overlays.is_active() || !notifications.is_window_focused() || recent_topics.is_visible()) {
```

Comment:
This is probably worth extracting as its own prep commit, since it's fixing a structural/logic bug.  Though I think structurally we'll want to rewrite this as calling a `message_viewport.is_visible()` function.